src/ray/pubsub/publisher.h

```diff
@@ -40,16 +40,25 @@ using PublisherID = UniqueID;
 
 namespace pub_internal {
 
+// Protobuf messages fail to serialize if 2GB or larger. Cap published messages
+// to 1GiB to ensure that we can publish the message. Individual messages
+// larger than this limit will be dropped.
+// TODO(swang): Pubsub clients should also ensure that they don't try to
+// publish messages larger than this.
+constexpr int64_t kMaxPubMessageSizeBytes = 1 << 30;
```

Comment:
This should be aligned with max gRPC payload size we're currently setting

Reply:
Actually i think we should just use that setting as the max message size

https://github.com/ray-project/ray/blob/f38582fd7637e8835f4ef86aaec9436290e916a8/src/ray/common/ray_config_def.h#L211C21-L211C42